x/programs/rust/sdk_macros/tests/compile.rs

```diff
@@ -0,0 +1,5 @@
+#[test]
+fn ui() {
+    let t = trybuild::TestCases::new();
+    t.compile_fail("tests/ui/**/*.rs");
```

Comment:
nit: not sure that "ui" is the best name here. maybe compilation_errors or something slightly more descriptive?

Reply:
+1

Reply:
Haha, I agree, but I also don't want to break convention:
https://docs.rs/trybuild/latest/trybuild/
https://github.com/serde-rs/serde/blob/master/test_suite/tests/compiletest.rs

Reply:
Going to keep this. I don't imagine too many program authors will look at the macro-tests. I would rather optimize for a Rust macro author